packages/template-retail-react-app/app/static/dwac-21.7.js

```diff
@@ -0,0 +1,470 @@
+// Enhance the dw.ac namespace
+(function(dw) {
+    if (typeof dw.ac === "undefined") {
+        return; // don't do anything if there was no <isactivedatahead> tag, that would create the ac namespace.
+    }
+    // Returns the value of the first cookie found whose name is accepted by the given function
+    function getCookieValue(/*function(x)*/ acceptFunction) {
+        var cookiePairs = document.cookie.split(';');
+        for (var i = 0; i < cookiePairs.length; i++)
+        {
+            var index = cookiePairs[i].indexOf('=');
+            if (index != -1) {
+                var name = trim(cookiePairs[i].substring(0, index));
+                if (acceptFunction(name)) {
+                    return trim(unescape(cookiePairs[i].substring(index + 1)));
+                }
+            }
+        }
+
+        return null;
+    };
+
+    // trims a string, replacing the jquery.trim
+    function trim(/*string*/value) {
+        return value.replace(/^\s\s*/, '').replace(/\s\s*$/, '');
+    };
+
+    // Sets a cookie with the given name and value
+    function setCookieValue(/*string*/ name, /*string*/ value, /*integer*/ millisToExpiry) {
+        var cookie = name + '=' + escape(value) + ';path=/';
+        if (millisToExpiry != -1) {
+            cookie += ";expires=" + expires.toGMTString();
+        }
+        document.cookie = cookie;
+    };
+
+    // URL encodes the given string to match the Java URLEncoder class
```

Comment:
There's no library for this as a standard encoding format? I'm definitely concerned about adding a very custom `urlencode` helper function like this especially if it's a standard Java encoding.

Reply:
> There's no library for this as a standard encoding format? I'm definitely concerned about adding a very custom `urlencode` helper function like this especially if it's a standard Java encoding.

@bfeister encodeURIComponent() JavaScript function would produce similar results https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent